tools/roadmap/validate_dag.py

```
@@ -0,0 +1,68 @@
+        if m:
+            nodes.add((m.group(1), int(m.group(2))))
+        m = re.match(r"\s*(N\d+)\s*==>\s*(N\d+)", ln)
+        if m:
+            edges_hard.append((m.group(1), m.group(2)))
+        m = re.match(r"\s*(N\d+)\s*-.->\s*(N\d+)", ln)
+        if m:
+            edges_soft.append((m.group(1), m.group(2)))
+    return nodes, edges_hard, edges_soft
+
+def gh_json(args):
+    out = run(["gh"] + args)
```

Ruff (0.14.3)

Consider `["gh", *args]` instead of concatenation

Replace with `["gh", *args]`

(RUF005)